### PR TITLE
feat(sponnet): jsonnet libs for v2 MPTs.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

**Comment by jtk54** [comment]
@kskewes PTAL.

File: sponnet/demo/demo-v2-config.jsonnet [hunk 1]
```diff
@@ -0,0 +1,16 @@
+local pipelines = import '../pipeline.libsonnet';
+
+local artifact = pipelines.artifact('front50/pipelineTemplate', '')
```
Comment (kskewes) [inline sponnet/demo/demo-v2-config.jsonnet]: The other artifact types (docker/gitlab/etc) are instantiated with `artifacts.type()`. 
I don't know if we would want to have `pipelineTemplate` as a separate `type()` or just leverage the underlying `artifact()` function directly as you have done which has all the required fields. We don't assert input correctness anyway.
I think you have a much better understanding of current and future plans so defer to you. :)

Comment (jtk54) [reply sponnet/demo/demo-v2-config.jsonnet]: That does make sense, I'll add in an explicit `front50/pipelineTemplate` type.

Comment (jtk54) [reply sponnet/demo/demo-v2-config.jsonnet]: Done.

**Pull request merged**